nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-common/src/main/java/org/apache/nifi/processors/grpc/FlowFileIngestService.java

```diff
@@ -102,23 +100,17 @@ public void send(final org.apache.nifi.processors.grpc.FlowFileRequest request,
         final ProcessSession session = sessionFactory.createSession();
 
         // if there's no space available, reject the request.
-        final long n = filesReceived.getAndIncrement() % FILES_BEFORE_CHECKING_DESTINATION_SPACE;
-        if (n == 0 || !spaceAvailable.get()) {
-            if (context.getAvailableRelationships().isEmpty()) {
-                spaceAvailable.set(false);
-                final String message = "Received request from " + remoteHost + " but no space available; Indicating Service Unavailable";
-                if (logger.isDebugEnabled()) {
-                    logger.debug(message);
-                }
-                final FlowFileReply reply = replyBuilder.setResponseCode(FlowFileReply.ResponseCode.ERROR)
-                        .setBody(message)
-                        .build();
-                responseObserver.onNext(reply);
-                responseObserver.onCompleted();
-                return;
-            } else {
-                spaceAvailable.set(true);
+        if (backpressureChecker.isBackpressure()) {
+            final String message = "Received request from " + remoteHost + " but no space available; Indicating Service Unavailable";
+            if (logger.isDebugEnabled()) {
+                logger.debug(message);
             }
```

Comment:
The conditional `logger.isDebugEnabled()` can be removed since the message is already computed. There is no cost savings to checking whether debug is enabled at this point.